lib/cylc/task_job_mgr.py

```diff
@@ -285,33 +285,45 @@ def submit_task_jobs(self, suite, itasks, is_simulation=False):
             cmd.append('--')
             cmd.append(glbl_cfg().get_derived_host_item(
                 suite, 'suite job log directory', host, owner))
-            stdin_file_paths = []
-            job_log_dirs = []
-            for itask in sorted(itasks, key=lambda itask: itask.identity):
-                if remote_mode:
-                    stdin_file_paths.append(
-                        get_task_job_job_log(
-                            suite, itask.point, itask.tdef.name,
-                            itask.submit_num))
-                job_log_dirs.append(get_task_job_id(
-                    itask.point, itask.tdef.name, itask.submit_num))
-                # The job file is now (about to be) used: reset the file write
-                # flag so that subsequent manual retrigger will generate a new
-                # job file.
-                itask.local_job_file_path = None
-                itask.state.reset_state(TASK_STATUS_READY)
-                if itask.state.outputs.has_custom_triggers():
-                    self.suite_db_mgr.put_update_task_outputs(itask)
-            cmd += job_log_dirs
-            self.proc_pool.put_command(
-                SubProcContext(
-                    self.JOBS_SUBMIT,
-                    cmd,
-                    stdin_file_paths=stdin_file_paths,
-                    job_log_dirs=job_log_dirs,
-                    **kwargs
-                ),
-                self._submit_task_jobs_callback, [suite, itasks])
+            # Chop itasks into a series of shorter lists if it's very big
+            # to prevent overloading of stdout and stderr pipes.
+            itasks = sorted(itasks, key=lambda itask: itask.identity)
+            chunk_size = len(itasks) // ((len(itasks) // 100) + 1) + 1
+            itasks_batches = [
+                itasks[i:i + chunk_size] for i in range(0,
+                                                        len(itasks),
+                                                        chunk_size)]
+            for i, itasks_batch in enumerate(itasks_batches):
+                debug_msg = "Submitting jobs batch {0}/{1} ({2} task(s))"
+                LOG.debug(debug_msg.format(i + 1,
+                                           len(itasks_batches),
+                                           len(itasks_batch)))
```

Comment:
I think my point above still holds true - it is more useful to have a debug statement that includes the leading portion of the command.

It is also best to call the `.debug` method with format string with the format operator and a list a arguments - as intended by the default usage of the method.
See also: https://docs.python.org/3/library/logging.html#logging.Logger.debug